1.1.0 kubernetes panic: send on closed channel

This has happened twice (2) in the last hour since i upgraded to 1.1.0 in our development cluster. 

The last log message before the panic was:
```
time="2016-11-22T14:43:00Z" level=error msg="Kubernetes connection error failed to decode watch event: GET \"https://127.0.0.1:9443/apis/extensions/v1beta1/ingresses\" : unexpected EOF, retrying in 639.617334ms" 
```

```
panic: send on closed channel

goroutine 4420 [running]:
panic(0xd292c0, 0xc4203f96a0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/containous/traefik/provider/k8s.(*clientImpl).watch.func1.1(0xc42050e900, 0xc420141260, 0xc420221080, 0x26, 0xc420141200)
	/go/src/github.com/containous/traefik/provider/k8s/client.go:280 +0x33f
created by github.com/containous/traefik/provider/k8s.(*clientImpl).watch.func1
	/go/src/github.com/containous/traefik/provider/k8s/client.go:286 +0xbf
```

I run traefik like this:
```
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  name: traefik-ingress-controller
  namespace: kube-system
  labels:
    app: traefik-ingress-lb
spec:
  template:
    metadata:
      labels:
        k8s-app: traefik-ingress-lb
        name: traefik-ingress-lb
    spec:
      terminationGracePeriodSeconds: 30
      hostNetwork: true
      containers:
      - image: traefik:v1.1.0
        name: traefik-ingress-lb
        imagePullPolicy: Always
        resources:
          limits:
            cpu: 1000m
            memory: 500Mi
          requests:
            cpu: 200m
            memory: 100Mi
        ports:
        - containerPort: 80
          hostPort: 80
        - containerPort: 8081
        args:
        - --retry.attempts=8
        - --retry
        - --web
        - --web.address=:8081
        - --kubernetes
        - --kubernetes.endpoint=https://127.0.0.1:9443
```

--- a/docs/toml.md
+++ b/docs/toml.md
@@ -511,7 +511,7 @@ address = ":8080"
 # To enable digest auth on the webui
 # with 2 user/realm/pass: test:traefik:test and test2:traefik:test2
 # You can use htdigest to generate those ones
-#   [web.auth.basic]
+#   [web.auth.digest]
 #     users = ["test:traefik:a2688e031edb4be6a3797f3882655c05 ", "test2:traefik:518845800f9e2bfb1f1f740ec24f074e"]
 
 ```

--- a/provider/k8s/client.go
+++ b/provider/k8s/client.go
@@ -7,6 +7,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/safe"
 	"github.com/parnurzeal/gorequest"
 	"net/http"
 	"net/url"
@@ -160,7 +161,7 @@ func (c *clientImpl) WatchAll(labelSelector string, stopCh <-chan bool) (chan in
 	if err != nil {
 		return watchCh, errCh, fmt.Errorf("failed to create watch: %v", err)
 	}
-	go func() {
+	safe.Go(func() {
 		defer close(watchCh)
 		defer close(errCh)
 		defer close(stopIngresses)
@@ -188,7 +189,7 @@ func (c *clientImpl) WatchAll(labelSelector string, stopCh <-chan bool) (chan in
 				watchCh <- event
 			}
 		}
-	}()
+	})
 
 	return watchCh, errCh, nil
 }
@@ -268,11 +269,16 @@ func (c *clientImpl) watch(url string, labelSelector string, stopCh <-chan bool)
 		return watchCh, errCh, fmt.Errorf("failed to do watch request: GET %q: %v", url, err)
 	}
 
-	go func() {
+	safe.Go(func() {
+		EndCh := make(chan bool, 1)
 		defer close(watchCh)
 		defer close(errCh)
-		go func() {
+		defer close(EndCh)
+		safe.Go(func() {
 			defer res.Body.Close()
+			defer func() {
+				EndCh <- true
+			}()
 			for {
 				var eventList interface{}
 				if err := json.NewDecoder(res.Body).Decode(&eventList); err != nil {
@@ -283,11 +289,12 @@ func (c *clientImpl) watch(url string, labelSelector string, stopCh <-chan bool)
 				}
 				watchCh <- eventList
 			}
-		}()
+		})
 		<-stopCh
-		go func() {
+		safe.Go(func() {
 			cancel() // cancel watch request
-		}()
-	}()
+		})
+		<-EndCh
+	})
 	return watchCh, errCh, nil
 }

--- a/traefik.sample.toml
+++ b/traefik.sample.toml
@@ -294,7 +294,7 @@
 # To enable digest auth on the webui
 # with 2 user/realm/pass: test:traefik:test and test2:traefik:test2
 # You can use htdigest to generate those ones
-#   [web.auth.basic]
+#   [web.auth.digest]
 #     users = ["test:traefik:a2688e031edb4be6a3797f3882655c05 ", "test2:traefik:518845800f9e2bfb1f1f740ec24f074e"]
 
 

